clang/include/clang/CIR/Interfaces/CIROpInterfaces.td

```diff
@@ -21,9 +21,24 @@ let cppNamespace = "::cir" in {
   // The CIRCallOpInterface must be used instead of CallOpInterface when looking
   // at arguments and other bits of CallOp. This creates a level of abstraction
   // that's useful for handling indirect calls and other details.
-  def CIRCallOpInterface : OpInterface<"CIRCallOpInterface", []> {
+  def CIRCallOpInterface : OpInterface<"CIRCallOpInterface", [CallOpInterface]> {
     // Currently we don't have any methods defined in CIRCallOpInterface. We'll
     // add more methods as the upstreaming proceeds.
+    let methods = [
+      InterfaceMethod<"", "mlir::Operation::operand_iterator",
+                      "arg_operand_begin", (ins)>,
+      InterfaceMethod<"", "mlir::Operation::operand_iterator",
+                      "arg_operand_end", (ins)>,
```

Comment:
@bcardosolopes any reason why this is part of interface and not just normal method on ops as is in `func::CallOp`?

Interface methods and methods in generally should not use snake_case, but camelCase.
Though it is unfortunately  named using a snake_case in other mlir dialects.

Reply:
> any reason why this is part of interface and not just normal method on ops as is in func::CallOp?

My recollection is that there was a problem with inherited interfaces and that wouldn't work. All we need here is the ability to override those differently from the inherited interface. If it works without the explicit declaration here I'm all for it!